docs/dev/background/ci.adoc

```diff
@@ -1,35 +1,102 @@
+
+* Quality report (sonarqube)
+    ** SonarQube is a platform for continuous inspection of code quality to perform automatic reviews with static analysis of code to detect bugs and code smells. It offers reports on duplicated code, coding standards, unit tests, code coverage, code complexity, comments, bugs, and security recommendations.
+    ** It renders the reports from pylint, flake8, bandit and code coverage in a more graphical way.
+    ** Reports are displayed at: https://sonarqube.ops-csc.com/dashboard?branch=develop&id=RS-PYTHON_rs-server_AYw0m7ixvQv-JMsowILQ
+
+== *Publish wheels and Docker images* workflow
+
+This workflow is run either automatically after adding a git tag to a branch, or manually. It:
+
+* Builds source code into Python wheel packages.
+    ** Upload them into the Python registry.
+    ** Make them available for manual download as GitHub Actions artifacts.
+
+* Builds the project Docker images (that use the Python wheel packages).
+    ** Scan them with Trivy for security issues and update results into the GitHub Security tab: https://github.com/RS-PYTHON/rs-server/security/code-scanning
+    ** Upload them into the ghcr.io GitHub package registry: https://github.com/orgs/RS-PYTHON/packages
+
+The git tag name must conform to the following syntax examples:
+
+* `v2.0` is a version for a release with at least major updates.
+* `v2.1` is a version for a release without any major updates.
+* `v2.1rc1` is the version for the first 2.1 release-candidate.
+* `v1.0a3` is the version for sprint 3 features integration.
+* `v1.0a3+dev9afcfc1` is dev commit version during the sprint 3.
+
+The `Poetry dynamic versioning` python plugin is then used to determine automatically the wheels and Docker images version name. Note that it is slighty different from the git tag name:
+
+* `v` is removed.
+* For Docker images, `+` is replaced by `.`
+
+So we have e.g. the wheel filenames:
+
+* `rs_server-2.0-py3-none-any.whl`
+* `rs_server-2.1-py3-none-any.whl`
+* `rs_server-2.1rc1-py3-none-any.whl`
+* `rs_server-1.0a3-py3-none-any.whl`
+* `rs_server-1.0a3+dev9afcfc1-py3-none-any.whl`
+
+And the Docker images:
+
+* `ghcr.io/rs-python/rs-server:2.0`
+* `ghcr.io/rs-python/rs-server:2.1`
+* `ghcr.io/rs-python/rs-server:2.1rc1`
+* `ghcr.io/rs-python/rs-server:1.0a3`
+* `ghcr.io/rs-python/rs-server:1.0a3.dev9afcfc1`
+
+When running manually the workflow, the version name is determined automatically as:
+
+* `0.0.0.post1.dev0+<short-hash-commit>` for the wheels.
+* `0.0.0.post1.dev0.<short-hash-commit>` for the Docker images.
+
+e.g. `0.0.0.post1.dev0+216f7fa` and `0.0.0.post1.dev0.216f7fa`.
+
+But running manually the workflow is discouraged, it is preferred to generate the wheels and Docker images by creating a new git tag (to be discussed).
+
+== *Generate documentation* workflow
+
+This workflow is run either automatically after adding a git tag to a branch, or manually.
```

Comment:
This workflow is run either automatically after adding a git tag to a commit, or manually.